webpack-subresource-integrity-embroider/index.js

```diff
@@ -19,19 +21,22 @@ class SubresourceIntegrityPlugin {
           [...scriptElements, ...linkElements].map(async (element) => {
             // calculate integrity
             const hashAlgorith = "sha384";
-            const fileName =
-              element.tagName === "SCRIPT"
+            const assetLocation = element.tagName === "SCRIPT"
                 ? element.getAttribute("src")
                 : element.getAttribute("href");
+            // strip publishPath from locations
+            const fileName = assetLocation.replace(publicPath, '/');
 
-            if (fileName === "/ember-cli-live-reload.js") {
+            if (fileName.startsWith("http") || fileName === "/ember-cli-live-reload.js") {
               // ember-cli-live-reload.js does not exist on disk
+              // skip external resources
```

Comment:
I'm not sure if we should do that. This plugin is likely used to ensure that _all_ assets are protected against malicious changes using Subresource Integrity. This is even more important for assets from another domain. Assets hosted on another domain are more likely exposed to malicious changes. If hosted on the same domain, it is more likely that an attacker would have control over the `index.html` as well and could remove/adjust the `integrity` value as well.

I tend towards throwing a meaningful error in this case instead.

Reply:
We'd need a branch that handles external resources instead then, we shouldn't add hashes to external resources since we'd just be hashing the content as it exist at build so changes to the external resource would just slip on by.

Are you suggesting we throw an error if we see an external resource without an integreity hash supplied and never do the hashing ourselves?

Reply:
I think if we detect an external resource we could hash it and supply the hash in the error message, asking the user if they trust the resource today to add the hash manually to the script/link tag. Then in future builds we skip resources with a present hash.

Not every resource offers a hash when supplied. I'd need this for https://rsms.me/inter/download/.

Reply:
> Are you suggesting we throw an error if we see an external resource without an integreity hash supplied and never do the hashing ourselves?

Yes.

I guess technically we should skip every element, which already has an `integrity` attribute. And throw an error if we cannot generate one ourself. Only exception is the `ember-cli-live-reload.js` because we know that it is not meant to be shipped in production.

> I think if we detect an external resource we could hash it and supply the hash in the error message, asking the user if they trust the resource today to add the hash manually to the script/link tag.

I would prefer keeping such complexity out of this library. We should instead tell the consumer in the error message that they need to add an `integrity` attribute themselves for that resource.

Reply:
> I would prefer keeping such complexity out of this library. We should instead tell the consumer in the error message that they need to add an integrity attribute themselves for that resource.

Not everyone is going to know how to do that and it might just make them more likely to turn off SRI entirely, especially if they are pulling in multiple different external resources. One of the reasons to reach for a library like this is to ease the burden of doing the right thing. I don't think there is there a big difference between using `readFile` vs using `fetch` to get the content of the resource.

I'll create a PR, and if you still feel that way I'll limit it to just ouputting an error for external resources without a hash.

Reply:
I see this Webpack plugin primarily as an interim solution until Ember's build system allows using [webpack-subresource-integrity](https://www.npmjs.com/package/webpack-subresource-integrity). I would limit features to those also supported by that established solution.

> Not everyone is going to know how to do that

The error message is a great option teaching it in my opinion.